src/mbgl/map/tile_worker.cpp

```diff
@@ -65,22 +67,18 @@ TileParseResult TileWorker::parsePendingLayers() {
     // dependencies.
     for (auto it = pending.begin(); it != pending.end();) {
         auto& layer = *it->first;
-        auto& bucket = it->second;
+        auto bucket = dynamic_cast<SymbolBucket*>(it->second.get());
```

Comment:
Only symbol layers can be pending, so it should never be null. I'll change the types to enforce this more strongly.